### PR TITLE
AnimatedBuilder API Doc improvements

### LINKED ISSUE #24829: Request a "Sample in an app" for AnimatedBuilder
The sample code for AnimatedBuilder's API doc doesn't show users how to "pass the pre-built subtree as the child parameter" as advised in the Performance Optimizations section. We have a study participant who read the doc but ended up extracting the subtree to a separate widget and pass the widget into the Transform widget instead of the AnimatedBuilder widget. I think having a full example showing how the subtree can be passed into the `Spinner` widget  would have been helpful. 

Cc: @gspencergoog

### PR DESCRIPTION
**Pull request by goderbauer**
## Description

Makes the child argument a little more complicated to show that you can pass in a full subtree here. Adds the sample code as animated diagram.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/24829.

## Tests

I added the following tests:

None, doc change only.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/widgets/transitions.dart [hunk 1]
```diff
@@ -918,6 +926,8 @@ class DefaultTextStyleTransition extends AnimatedWidget {
 /// ```
 /// {@end-tool}
 ///
+/// ![A spinning container.](https://flutter.github.io/assets-for-api-docs/assets/widgets/aniamted_builder.mp4)
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/transitions.dart]: In review here: https://github.com/flutter/assets-for-api-docs/pull/107

Comment (Piinks) [reply packages/flutter/lib/src/widgets/transitions.dart]: This should use the `{animation...}` embedding format correct?

Comment (goderbauer) [reply packages/flutter/lib/src/widgets/transitions.dart]: Yes! Thanks for the catch!

File: packages/flutter/lib/src/widgets/transitions.dart [hunk 2]
```diff
@@ -918,6 +926,8 @@ class DefaultTextStyleTransition extends AnimatedWidget {
 /// ```
 /// {@end-tool}
 ///
+/// {@animation 300 300 https://flutter.github.io/assets-for-api-docs/assets/widgets/aniamted_builder.mp4}
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/transitions.dart]: Uber nit:
```suggestion
/// {@animation 300 300 https://flutter.github.io/assets-for-api-docs/assets/widgets/animated_builder.mp4}
```
It looks like this typo would need to be fixed in the assets repo first. It links to the proper asset so I think it's ok as is?

**Review by Piinks — APPROVED** [review]
LGTM! 
There is a minor typo, but I do not think it is relevant since it links to the proper resource.

File: packages/flutter/lib/src/widgets/transitions.dart (hunk 2 above)
Comment (goderbauer) [reply packages/flutter/lib/src/widgets/transitions.dart]: Fixing spelling here: https://github.com/flutter/assets-for-api-docs/pull/109

**Pull request merged**